scripts/zkai_native_seq32_attention_mlp_adapter_variant_selector_gate.py

```
@@ -0,0 +1,713 @@
+        "best_variant_fri_sample_overhang_bytes": 16,
+        "nanozk_reported_d128_block_proof_bytes": NANOZK_REPORTED_D128_BLOCK_PROOF_BYTES,
+        "current_champion_typed_ratio_to_nanozk_reported_row": "6.096812",
+        "proof_size_comparable_external_rows": 0,
+    }
+    if summary != expected:
+        raise NativeSeq32AdapterVariantSelectorGateError("summary drift")
+
+
+def validate_variants(variants: list[Any]) -> None:
+    if len(variants) != len(EXPECTED_ROWS):
+        raise NativeSeq32AdapterVariantSelectorGateError("variant inventory drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)